python/pyspark/tests/test_daemon.py

```diff
@@ -47,9 +47,12 @@ def do_termination_test(self, terminator):
         # daemon should accept connections
         self.assertTrue(self.connect(port))
 
+        # wait worker process spawned from daemon exit.
+        time.sleep(1)
+
         # request shutdown
         terminator(daemon)
-        time.sleep(1)
+        daemon.wait(5)
```

Comment:
Hi, Guys.

The `timeout` argument is Python 3 only feature.
- https://docs.python.org/2/library/subprocess.html#subprocess.Popen.wait
- https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait

Since this will break Python 2.7 test in all branches. I'll revert this commit from all the branches.